vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/ComponentQuery.java

```diff
@@ -465,70 +451,62 @@ public <E extends Component> ComponentQuery<E> thenOnFirst(
      */
     public <E extends Component> ComponentQuery<E> thenOn(int index,
             Class<E> componentType) {
-        return new ComponentQuery<>(componentType, wrapperFactory)
-                .from(atIndex(index).getComponent());
+        return new ComponentQuery<>(componentType).from(atIndex(index));
     }
 
     /**
-     * Executes the search against current context and returns the test wrapper
-     * for the result, expecting to find exactly one component.
+     * Executes the search against current context and returns the component,
+     * expecting to find exactly one.
      *
      * Exceptions are thrown if the search produces zero or more than one
      * result.
      *
-     * @return a test wrapper for the component of the type specified in the
-     *         constructor.
+     * @return the component of the type specified in the constructor.
      * @throws java.util.NoSuchElementException
      *             if not exactly one component is found
      */
     @SuppressWarnings("unchecked")
-    public <X extends ComponentTester<? extends T>> X single() {
+    public <X extends T> X single() {
         return (X) find();
```

Comment:
Yes, seems much cleaner.
The `<X extends T>` may only be useful if you search for a component and expect to have a subclass; for example query for TextField but you know the query will extract MyCustomTextField. But this may be a very limited use case, so the developer can do an explicit cast.

Reply:
I feel that in those cases they should just search for the actual component, so getting the type they search for feels cleaner and better.

Reply:
done